hello-world.md

```diff
@@ -66,7 +66,30 @@ func TestHello(t *testing.T) {
 }
 ```
 
-Before explaining, let's just run the code. Run `go test` in your terminal. It should've passed! Just to check, try deliberately breaking the test by changing the `want` string.
+## Go modules?
+
+The next step is to run the tests. Enter `go test` in your terminal. If the tests pass, then you are probably using an earlier version of Go. However, if you are using Go 1.16 or later, then the tests will likely not run at all. Instead, you will see an error message like this in the terminal:
+
+```shell
+$ go test
+go: cannot find main module; see 'go help modules'
+```
+
+What's the problem? In a word, [modules](https://blog.golang.org/go116-module-changes). Luckily, the problem is easy to fix. Enter `go mod init` in your terminal. That will create a new file with the following contents:
+
+```go
+module hello
+
+go 1.16
+```
+
+This file tells the `go` tools essential information about your code. If you planned to distribute your application, you would include where the code was available for download as well as information about dependencies. For now, your module file is minimal, and you can leave it that way. To read more about modules [you should check out the reference in the Golang documentation](https://golang.org/doc/modules/gomod-ref). We can get back to testing and learning go now since the tests should run, even on Go 1.16.
```

Comment:
```suggestion
This file tells the `go` tools essential information about your code. If you planned to distribute your application, you would include where the code was available for download as well as information about dependencies. For now, your module file is minimal, and you can leave it that way. To read more about modules, [you cab check out the reference in the Golang documentation](https://golang.org/doc/modules/gomod-ref). We can get back to testing and learning go now since the tests should run, even on Go 1.16.
```